python/pyiceberg/cli/console.py

```diff
@@ -0,0 +1,185 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+
+import click
+from rich.console import Console
+from rich.table import Table
+from rich.tree import Tree
+
+from pyiceberg.catalog.rest import RestCatalog
+from pyiceberg.exceptions import NamespaceNotEmptyError, NoSuchNamespaceError, NoSuchTableError
+
+
+@click.group()
+@click.option("--catalog", type=click.Choice(["rest"], case_sensitive=False), default="rest")
+@click.option("--uri", default=lambda: os.environ.get("PYICEBERG_URI"))
+@click.option("--credentials", default=lambda: os.environ.get("PYICEBERG_CREDENTIALS"))
+@click.pass_context
+def run(ctx, catalog, uri, credentials):
+    if catalog == "rest":
+        ctx.ensure_object(dict)
+        ctx.obj["catalog"] = RestCatalog(name="catalog", properties={}, uri=uri, credentials=credentials)
+
+
+@run.command()
+@click.pass_context
+def list_namespace(ctx):
```

Comment:
Minor: should be plural, `list_namespaces` to be consistent with `list_tables`.

Reply:
Thanks, but I'll combine namespaces and tables like you suggested, so this will go away :)